src/m365/outlook/commands/message/message-remove.spec.ts

```diff
@@ -0,0 +1,249 @@
+import sinon from 'sinon';
+import auth from '../../../../Auth.js';
+import commands from '../../commands.js';
+import request from '../../../../request.js';
+import { telemetry } from '../../../../telemetry.js';
+import { Logger } from '../../../../cli/Logger.js';
+import { CommandError } from '../../../../Command.js';
+import { pid } from '../../../../utils/pid.js';
+import { session } from '../../../../utils/session.js';
+import { sinonUtil } from '../../../../utils/sinonUtil.js';
+import { cli } from '../../../../cli/cli.js';
+import { accessToken } from '../../../../utils/accessToken.js';
+import command from './message-remove.js';
+
+describe(commands.MESSAGE_REMOVE, () => {
+  const messageId = 'AAMkAGRlM2Y5YTkzLWI2NzAtNDczOS05YWMyLTJhZGY2MGExMGU0MgBGAAAAAABIbfA8TbuRR7JKOZPl5FPxBwB8kpUvTuxuSYh8eqNsOdGBAAAAAAEMAAB8kpUvTuxuSYh8eqNsOdGBAADb58MCAAA=';
+  const userId = '6799fd1a-723b-4eb7-8e52-41ae530274ca';
+  const userPrincipalName = 'john.doe@contoso.com';  
+
+  let log: string[];
+  let logger: Logger;
+  let promptIssued: boolean;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.service.connected = true;
+    auth.service.accessTokens[auth.defaultResource] = {
+      expiresOn: 'abc',
+      accessToken: 'abc'
+    };
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    sinon.stub(cli, 'promptForConfirmation').callsFake(() => {
+      promptIssued = true;
+      return Promise.resolve(false);
+    });
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(false);
+    promptIssued = false;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.delete,
+      accessToken.isAppOnlyAccessToken,
+      cli.handleMultipleResultsFound,
+      cli.promptForConfirmation
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.service.connected = false;
+    auth.service.accessTokens = {};
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.MESSAGE_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('removes specific message using delegated permissions without prompting for confirmation', async () => {
+    const deleteRequestStub = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/me/messages/${messageId}`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { id: messageId, force: true, verbose: true } });
+    assert(deleteRequestStub.called);
```

Comment:
Let's prefer the more strict `.calledOnce`, let's do this for other tests as well.